### ISSUE TITLE
Enzyme Adapter not loading on click of individual test.

### ISSUE THREAD

**Issue by ellisio** [hbenl/vscode-test-explorer#34]
We are using Enzyme with our Mocha unit tests. I can run all tests using the play button the Test Explorer UI. If I go to click on an actual test from the tree, I get the following error in the Test Explorer output:

```
Error: 
      Enzyme Internal Error: Enzyme expects an adapter to be configured, but found none.
      To configure an adapter, you should call `Enzyme.configure({ adapter: new Adapter() })`
      before using any of Enzyme's top level APIs, where `Adapter` is the adapter
      corresponding to the library currently being tested. For example:

      import Adapter from 'enzyme-adapter-react-15';

      To find out more about this, see http://airbnb.io/enzyme/docs/installation/index.html
    
    at validateAdapter (node_modules/enzyme/build/validateAdapter.js:16:11)
    at getAdapter (node_modules/enzyme/build/getAdapter.js:27:36)
    at new ReactWrapper (node_modules/enzyme/build/ReactWrapper.js:126:49)
    at mount (node_modules/enzyme/build/mount.js:21:10)
    at mount (packages/arcadia-ui/src/test-utils/createMount.js:12:53)
    at Context.mount (packages/arcadia-ui/src/styles/withStyles.test.js:318:9)
```

I have the following in `<projectRoot>/test/mocha.opts`:

```
--require @babel/register
--require jsdom-global/register
--reporter dot
--recursive
test/utils/setup.js
```

Our `<projectRoot>/test/utils/setup.js` contains:

```
/* eslint-disable no-console */
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Set to true for Mocha.
process.browser = true;

// Configure console.error calls.
console.error = (...args) => {
  console.info(...args);
  throw new Error(...args);
};

// Attach the Enzyme Adapter
configure({ adapter: new Adapter() });
```

It appears the root call to run the tests is using Mocha as expected, but direct calls are not. Let me know if I can provide any other information (not really sure what else to add at this time).

**Comment by hbenl** [comment]
* are you using the `mochaExplorer.pruneFiles` option?
* what happens if you try to run a single test using Mocha's `--grep` option on the command line (e.g. `mocha --grep 'My test'`)?
* please create a logfile (using `mochaExplorer.logfile`) for a successful and an unsuccessful test run and attach them here

**Comment by ellisio** [comment]
@hbenl To answer you questions:

1. No, `mochaExplorer.pruneFiles` is false.

2. Mocha output:

```
$ npx mocha --grep 'Avatar renders a div'


  0 passing (2ms)

$ npx mocha 'packages/**/*.test.js' --exclude '**/node_modules/**' --grep 'Avatar renders a div'

  ✓ arcadia-ui/Avatar/Avatar renders a div: 4ms

  1 passing (21ms)
```

3. Log output:

```
[2019-02-11 15:52:14.469] [INFO] Configuration changed
[2019-02-11 15:52:23.060] [INFO] Running test(s) ["/Users/andrew/Developer/salesmessage/arcadia/packages/arcadia-ui/src/Avatar/Avatar.test.js: arcadia-ui/Avatar/Avatar renders a div"] of /Users/andrew/Developer/salesmessage/arcadia
[2019-02-11 15:52:23.060] [DEBUG] Looking for mocha options in /Users/andrew/Developer/salesmessage/arcadia/test/mocha.opts
[2019-02-11 15:52:23.060] [DEBUG] Options from mocha.opts file: {"requires":["@babel/register","jsdom-global/register"]}
[2019-02-11 15:52:23.060] [DEBUG] Globs from mocha.opts file: ["test/setup.js"]
[2019-02-11 15:52:23.060] [DEBUG] Files from mocha.opts file: []
[2019-02-11 15:52:23.060] [DEBUG] Looking for test files ["packages/**/*.test.js"] in /Users/andrew/Developer/salesmessage/arcadia
[2019-02-11 15:52:23.116] [DEBUG] Found test files ["/Users/andrew/Developer/salesmessage/arcadia/packages/arcadia-ui-styles/src/index.test.js","/Users/andrew/Developer/salesmessage/arcadia/packages/arcadia-ui/src/index.test.js","/Users/andrew/Developer/salesmessage/arcadia/packages/arcadia-ui/src/Box/Box.test.js","/Users/andrew/Developer/salesmessage/arcadia/packages/arcadia-ui/src/Button/Button.test.js","/Users/andrew/Developer/salesmessage/arcadia/packages/arcadia-ui/src/Avatar/Avatar.test.js","/Users/andrew/Developer/salesmessage/arcadia/packages/arcadia-ui/node_modules/@salesmessage/arcadia-ui-styles/src/index.test.js"]
[2019-02-11 15:52:23.127] [DEBUG] Using nodePath: /usr/local/opt/node@10/bin/node
[2019-02-11 15:52:23.128] [DEBUG] Using Mocha options: {"ui":"bdd","timeout":2000,"retries":0,"requires":["@babel/register","jsdom-global/register"],"exit":false}
[2019-02-11 15:52:23.128] [DEBUG] Using working directory: /Users/andrew/Developer/salesmessage/arcadia
[2019-02-11 15:52:23.128] [DEBUG] Using environment variable config: {}
[2019-02-11 15:52:23.298] [INFO] Worker: Trying require('@babel/register')
[2019-02-11 15:52:23.378] [INFO] Worker: Trying require('jsdom-global/register')
[2019-02-11 15:52:23.968] [INFO] Worker: Running tests
[2019-02-11 15:52:24.378] [INFO] Received {"type":"suite","suite":"undefined: ","state":"running"}
[2019-02-11 15:52:24.390] [INFO] Received {"type":"suite","suite":"/Users/andrew/Developer/salesmessage/arcadia/packages/arcadia-ui/src/Avatar/Avatar.test.js: arcadia-ui/Avatar/Avatar","state":"running"}
[2019-02-11 15:52:24.390] [INFO] Received {"type":"test","test":"/Users/andrew/Developer/salesmessage/arcadia/packages/arcadia-ui/src/Avatar/Avatar.test.js: arcadia-ui/Avatar/Avatar renders a div","state":"running"}
[2019-02-11 15:52:24.399] [INFO] Received {"type":"test","test":"/Users/andrew/Developer/salesmessage/arcadia/packages/arcadia-ui/src/Avatar/Avatar.test.js: arcadia-ui/Avatar/Avatar renders a div","state":"failed","message":"Error: \n      Enzyme Internal Error: Enzyme expects an adapter to be configured, but found none.\n      To configure an adapter, you should call `Enzyme.configure({ adapter: new Adapter() })`\n      before using any of Enzyme's top level APIs, where `Adapter` is the adapter\n      corresponding to the library currently being tested. For example:\n\n      import Adapter from 'enzyme-adapter-react-15';\n\n      To find out more about this, see http://airbnb.io/enzyme/docs/installation/index.html\n    \n    at validateAdapter (node_modules/enzyme/build/validateAdapter.js:16:11)\n    at getAdapter (node_modules/enzyme/build/getAdapter.js:27:36)\n    at new ShallowWrapper (node_modules/enzyme/build/ShallowWrapper.js:193:47)\n    at shallow (node_modules/enzyme/build/shallow.js:21:10)\n    at Context.<anonymous> (packages/arcadia-ui/src/Avatar/Avatar.test.js:8:21)","decorations":[{"line":7,"message":"\n      Enzyme Internal Error: Enzyme expects an adapter to be configured, but found none.\n      To configure an adapter, you should call `Enzyme.configure({ adapter: new Adapter() })`\n      before using any of Enzyme's top level APIs, where `Adapter` is the adapter\n      corresponding to the library currently being tested. For example:\n\n      import Adapter from 'enzyme-adapter-react-15';\n\n      To find out more about this, see http://airbnb.io/enzyme/docs/installation/index.html\n    "}]}
[2019-02-11 15:52:24.399] [INFO] Received {"type":"suite","suite":"/Users/andrew/Developer/salesmessage/arcadia/packages/arcadia-ui/src/Avatar/Avatar.test.js: arcadia-ui/Avatar/Avatar","state":"completed"}
[2019-02-11 15:52:24.399] [INFO] Received {"type":"suite","suite":"undefined: ","state":"completed"}
[2019-02-11 15:52:24.417] [INFO] Worker finished
```

I also have the following set:

```
"mochaExplorer.files": "packages/**/*.test.js",
```

If this is not set, Mocha Test explorer returns blank.

**Comment by ellisio** [comment]
I was able to fix this by modifying my `mocha.opts` to be the following and removing the `mochaExplorer.files` settings value:

```
--require @babel/register
--require jsdom-global/register
--reporter list
--recursive
test/setup.js
packages/**/*.test.js
```

**Comment by hbenl** [comment]
@ellisio Thanks for reporting your workaround, with this info I finally figured out what the initial problem was: you had your setup file in `mocha.opts` and the test files on the command line - mocha merges the files and loads all of them. Then you put your test files in the VS Code settings, but the adapter didn't merge the files from `mocha.opts` and the settings - instead, the files from the settings did override those from `mocha.opts`, so the setup file wasn't loaded.
I have fixed this in version 1.6.1: now the adapter merges the files from `mocha.opts` and `mochaExplorer.files`.
